cmd/oceantv/main.go

```diff
@@ -0,0 +1,232 @@
+	"github.com/ausocean/cloud/gauth"
+	"github.com/ausocean/cloud/notify"
+)
+
+const (
+	projectID          = "oceantv"
+	cronServiceAccount = "oceancron@appspot.gserviceaccount.com"
+	senderEmail        = "vidgrindservice@gmail.com" // TODO: Change this.
+	locationID         = "Australia/Adelaide"        // TODO: Use site location.
+)
+
+var (
+	setupMutex    sync.Mutex
+	settingsStore iotds.Store
+	mediaStore    iotds.Store
+	debug         bool
+	standalone    bool
+	notifier      notify.Notifier
+	cronSecret    []byte
+)
+
+func main() {
+	defaultPort := 8082
+	v := os.Getenv("PORT")
+	if v != "" {
+		i, err := strconv.Atoi(v)
+		if err == nil {
+			defaultPort = i
+		}
+	}
+
+	var host string
+	var port int
+	flag.BoolVar(&debug, "debug", false, "Run in debug mode.")
+	flag.BoolVar(&standalone, "standalone", false, "Run in standalone mode.")
+	flag.StringVar(&host, "host", "localhost", "Host we run on in standalone mode")
+	flag.IntVar(&port, "port", defaultPort, "Port we listen on in standalone mode")
+	flag.Parse()
+
+	// Perform one-time setup or bail.
+	setup(context.Background())
+
+	http.HandleFunc("/_ah/warmup", warmupHandler)
+	http.HandleFunc("/broadcast/", broadcastHandler)
+	http.HandleFunc("/", indexHandler)
+
+	log.Printf("Listening on %s:%d", host, port)
+	log.Fatal(http.ListenAndServe(fmt.Sprintf("%s:%d", host, port), nil))
+}
+
+// warmupHandler handles App Engine warmup requests. It simply ensures that the instance is loaded.
+func warmupHandler(w http.ResponseWriter, r *http.Request) {
+	indexHandler(w, r)
+}
+
+// indexHandler handles requests for the home page and is here just to
+// test that the service is running. Devices do not use this endpoint.
+func indexHandler(w http.ResponseWriter, r *http.Request) {
+	logRequest(r)
+	w.Write([]byte("OK"))
+}
+
+// setup executes per-instance one-time initialization. Any errors are
+// considered fatal.
+func setup(ctx context.Context) {
+	setupMutex.Lock()
+	defer setupMutex.Unlock()
+
+	if settingsStore != nil {
+		return
+	}
+
+	var err error
+	if standalone {
+		log.Printf("Running in standalone mode")
+		settingsStore, err = iotds.NewStore(ctx, "file", projectID, "store")
+		if err != nil {
+			mediaStore = settingsStore
+		}
+	} else {
+		log.Printf("Running in App Engine mode")
+		settingsStore, err = iotds.NewStore(ctx, "cloud", "netreceiver", "")
+		if err != nil {
+			mediaStore, err = iotds.NewStore(ctx, "cloud", "vidgrind", "")
+		}
+	}
+	if err != nil {
+		log.Fatalf("could not set up datastore: %v", err)
+	}
+	iotds.RegisterEntities()
+
+	cronSecret, err = gauth.GetHexSecret(ctx, projectID, "cronSecret")
+	if err != nil || cronSecret == nil {
+		log.Printf("could not get cronSecret: %v", err)
+	}
+
+	err = notifier.Init(ctx, projectID, senderEmail, &timeStore{})
+	if err != nil {
+		log.Fatalf("could not set up email notifier: %v", err)
+	}
+}
+
+// timeStore implements a notify.TimeStore that uses iotds.Variable for persistence.
+type timeStore struct {
```

Comment:
Has gofmt been run ? I feel like it would have made this 
`type timeStore struct {}`

Reply:
Yes, gofmt has been run and it was happy.